pallets/msa/src/lib.rs

```diff
@@ -2883,7 +2904,7 @@ impl<T: Config> TargetValidator for Pallet<T> {
 	}
 }
 
-impl<T: Config> SchemaGrantValidator<BlockNumberFor<T>> for Pallet<T> {
+impl<T: Config> GrantValidator<IntentId, BlockNumberFor<T>> for Pallet<T> {
```

Comment:
Should this use Self::BlockNumber?

```suggestion
impl<T: Config> GrantValidator<IntentId, Self::BlockNumber> for Pallet<T> {
```